src/createQueryClient.ts

```diff
@@ -40,9 +47,68 @@ export function createQueryClient(options?: ClientOptions): QueryClient {
     }
   }
 
+  const setQueryData: SetQueryData = (
+    param1: QueryTag | QueryTag[] | QueryAction, 
+    param2: Parameters<QueryAction> | QueryDataSetter, 
+    param3?: QueryDataSetter
```

Comment:
trying this out rather than the `queryTagOrAction, parametersOrSetter, maybeSetter` pattern. Not set on it, but kinda easier to follow I think